homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,229 @@
+        for port in list_of_ports:
+            ports_as_string.append(
+                f"{port.device}"
+                + f" - {port.product}"
+                + f" - {format(port.vid, 'x')}:{format(port.pid, 'x')}"
+            )
+        ports_as_string.append(MANUAL_PATH)
+        ports_as_string.append(REFRESH_LIST)
+
+        if user_input is not None:
+            if user_input.get(CONF_UNIQUE_ID) is not None:
+                # provided unique id from entry that needs to be updated
+                self.existing_entry_id = user_input.get(CONF_UNIQUE_ID)
+
+            if user_input.get(CONF_USB_PATH) is not None:
+                selection = user_input[CONF_USB_PATH]
+
+                # no usb should be used, finish flow
+                if selection == DONT_USE_USB:
+                    return self.async_finish_flow()
+
+                # show a form with text field to enter manual path
+                if selection == MANUAL_PATH:
+                    return await self.async_step_usb_manual_config()
+
+                # user can refresh the list while in this step
+                if selection != REFRESH_LIST:
+                    # get serial-id info
+                    selected_port: ListPortInfo = list_of_ports[
+                        (ports_as_string.index(selection) - 1)
+                    ]
+                    self.usb_path = await self.hass.async_add_executor_job(
+                        get_serial_by_id, selected_port.device
+                    )
+
+                    # check if we are updating an existing entry
+                    if self.existing_entry_id is not None:
+                        return await self.async_update_existing_entry_usb_path()
+
+                    return self.async_finish_flow()
+
+        return self.async_show_form(
+            step_id="usb_config",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USB_PATH): vol.In(ports_as_string)}
+            ),
+        )
+
+    async def async_step_usb_manual_config(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manually enter Crownstone USB dongle path."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="usb_manual_config",
+                data_schema=vol.Schema({vol.Required(CONF_USB_MANUAL_PATH): str}),
+            )
+
+        # save usb path
+        self.usb_path = user_input[CONF_USB_MANUAL_PATH]
+
+        # check if we are updating an existing entry
+        if self.existing_entry_id is not None:
+            return await self.async_update_existing_entry_usb_path()
+
+        return self.async_finish_flow()
+
+    async def async_update_existing_entry_usb_path(self) -> FlowResult:
+        """Update usb path of existing entry."""
+        existing_entry = self.hass.config_entries.async_get_entry(
+            str(self.existing_entry_id)
+        )
+        if existing_entry is not None:
```

Comment:
Reverse the condition and outdent the block below

```suggestion
        if existing_entry is None:
            # if we couldn't find the entry
            return self.async_abort(reason="usb_setup_unsuccessful")        
```